outetts.js/version/v1/interface.js

```diff
@@ -0,0 +1,275 @@
+        /* RIFF chunk length */
+        view.setUint32(4, 36 + samples.length * 4, true);
+        /* RIFF type */
+        writeString(view, 8, "WAVE");
+        /* format chunk identifier */
+        writeString(view, 12, "fmt ");
+        /* format chunk length */
+        view.setUint32(16, 16, true);
+        /* sample format (raw) */
+        view.setUint16(20, 3, true);
+        /* channel count */
+        view.setUint16(22, 1, true);
+        /* sample rate */
+        view.setUint32(24, sr, true);
+        /* byte rate (sample rate * block align) */
+        view.setUint32(28, sr * 4, true);
+        /* block align (channel count * bytes per sample) */
+        view.setUint16(32, 4, true);
+        /* bits per sample */
+        view.setUint16(34, 32, true);
+        /* data chunk identifier */
+        writeString(view, 36, "data");
+        /* data chunk length */
+        view.setUint32(40, samples.length * 4, true);
+
+        for (let i = 0; i < samples.length; ++i, offset += 4) {
+            view.setFloat32(offset, samples[i], true);
+        }
+
+        return buffer;
+    }
+
+    async save(path) {
+        const wave = this.to_wav();
+        if (typeof window !== "undefined") {
+            // Create blob
+            const blob = new Blob([this.to_wav()], { type: "audio/wav" });
+            const url = URL.createObjectURL(blob);
+            const a = document.createElement("a");
+            a.href = url;
+            a.download = path;
+            a.click();
+
+            // Clean up
+            URL.revokeObjectURL(url);
+        } else {
+            // Dynamically import fs
+            const fs = await import("fs");
+            fs.writeFileSync(path, Buffer.from(wave));
+        }
+    }
+}
+
+export class InterfaceHF {
+    /**
+     *
+     */
+    constructor({ config, audio_codec, prompt_processor, model }) {
+        this.language = config.language;
+        this.languages = config.languages;
+
+        /** @param {HFModelConfig} */
+        this.config = config;
+
+        /** @type {AudioCodec} */
+        this.audio_codec = audio_codec;
+
+        /** @type {PromptProcessor} */
+        this.prompt_processor = prompt_processor;
+
+        /** @type {HFModel} */
+        this.model = model;
+    }
+
+    /**
+     * @param {HFModelConfig} config
+     */
+    static async load(config) {
+        const prompt_processor = new PromptProcessor(
+            await AutoTokenizer.from_pretrained(config.tokenizer_path ?? config.model_path),
+            config.languages,
+        );
+
+        const model = new HFModel(
+            await AutoModelForCausalLM.from_pretrained(config.model_path, {
+                device: config.device,
+                dtype: config.dtype,
+            }),
+        );
+
+        const audio_codec = new AudioCodec(
+            await PreTrainedModel.from_pretrained("onnx-community/WavTokenizer-large-speech-75token_decode", {
+                device: config.device,
```

Comment:
```suggestion
                device: config.device,
```
[TODO] Support different devices/dtypes for audio codec and language model